### ISSUE TITLE
mask_zero should be set to True in the embedding layer

### ISSUE THREAD

**Issue by doctorpangloss** [yxtay/char-rnn-text-generation#3]
https://github.com/yxtay/char-rnn-text-generation/blob/216f3683100047d70f856bb122a27eafbe62c7bc/keras_model.py#L32

**Comment by yxtay** [comment]
Hi @doctorpangloss, thank you for the suggestion.

Took a brief look at the [documentation](https://keras.io/layers/embeddings/#embedding). The argument `mask_zero` should be used if 0 is used as a special padding value. This is not relevant in the way the data is prepared in the code. The character sequences are of equal length and, hence, no padding was required.

In the code, the index 0 can be seen as being used for unknown tokens. In the preprocessing, unknown tokens are converted to empty string and hence removed. In inference, any prediction of 0 will be represented as an empty string and not be included in the predicted sequence. The index 0 does not appear in the training dataset, so the prediction of 0 should not happen.

Since the default value of `mask_zero` is `false`, it doesn't have to be specified in this use case.

Hope that is clear.
